### ISSUE TITLE
`reportPossiblyUnboundVariable` false negative on assignments to instance attribute

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#639]
Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.13&typeCheckingMode=all&reportUnusedExpression=false&code=MYGwhgzhAEBiD28BcAoa7oBMCmAzaA%2BgQJYB2xALkQBQTYi4CU0AtAHzQBy8p2qGA6MXwAjRCGqN%2BgmWGgBeaAEY0M6GFUzh0MfAlTNa9HQYA6OYpVGTuc0A)

```python
class Foo:
    def __init__(self) -> None:
        if bool():
            a = 1
        a # "a" is possibly unbound  (reportPossiblyUnboundVariable)
        if bool():
            self.a = 1
        self.a # no error
```

**Comment by DetachHead** [comment]
maybe related?: #682
